src/__tests__/ProductView/ProductView.js

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import ProductView from '../../ProductView/ProductView.js';
+import renderer from 'react-test-renderer';
+import TestProduct from './../testData/product.js'
+import TestEventTypes from './../testData/eventTypes.js'
+import TestEvents from './../testData/events.js'
+let instance;
+
+test("Rendering of ProductView", () => {
+    const mockCallback = jest.fn();
+    const component = renderer.create(
+        <ProductView
+            ref={(child) => {instance = child}}
+            dataSource={{fetchProduct: mockCallback, fetchEventTypesOf: mockCallback, fetchEventsOf: mockCallback}}
+            params={{productId: 0}}/>
+    );
+    instance.handleProductData(TestProduct.PRODUCT);
```

Comment:
Nice one using mockCallback!
But I would suggest to generate an independent test for each of the functions.